packages/database/data-migrations/20241120150728_product_revamp/data-migration.ts

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable no-console -- logging is allowed in migration scripts */
+import { Prisma, PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+const TRANSACTION_TIMEOUT = 3 * 60 * 1000; // 3 minutes in milliseconds
+
+type Plan = "free" | "startup" | "scale" | "enterprise";
+
+const projectsLimitByPlan: Record<Plan, number | null> = {
+  free: 3,
+  startup: 3,
+  scale: 5,
+  enterprise: null,
+};
+
+async function runMigration(): Promise<void> {
+  const startTime = Date.now();
+  console.log("Starting data migration...");
+
+  await prisma.$transaction(
+    async (transactionPrisma) => {
+      const organizations = await transactionPrisma.organization.findMany({
+        where: {
+          billing: {
+            path: ["limits", "projects"],
+            equals: Prisma.DbNull,
+          },
+        },
+        select: {
+          id: true,
+          billing: true,
+        },
+      });
+
+      const updateOrganizationPromises = organizations.map((org) =>
+        transactionPrisma.organization.update({
```

Comment:
when running the data migration multiple times we update the enterprise customers multiple times since `"limits", "projects"` is null. Can we maybe add an additional check and only run the prisma update call if the value changed?

Reply:
I tried this already, and it works fine as there are two types of null values in postges `jsonNull` and `dbNull` and since we are checking inside a JSON object, where clause with `jsonNull` returns objects with `projects: null;` but `dbNull` returns records with inexistent `projects` values.
https://stackoverflow.com/a/71033046